politeiawww/cmd/politeiavoter/trickle.go

```diff
@@ -249,23 +250,23 @@ func (p *piv) voteTicket(ectx context.Context, bunchID, voteID, of int, va voteA
 
 				return nil
 			}
-		} else {
-			// Success, log it and exit
-			err = p.jsonLog(successJournal, va.Vote.Token, vr)
-			if err != nil {
-				return fmt.Errorf("3 jsonLog: %v", err)
-			}
+		}
 
-			// All done with this vote
-			// Vote completed
-			p.Lock()
-			p.ballotResults = append(p.ballotResults, *vr)
-			fmt.Printf("%v finished bunch %v vote %v -- "+
-				"total progress %v/%v\n", time.Now(), bunchID,
-				voteID, len(p.ballotResults), cap(p.ballotResults))
-			p.Unlock()
-			return nil
+		// Success, log it and exit
+		err = p.jsonLog(successJournal, va.Vote.Token, vr)
+		if err != nil {
+			return fmt.Errorf("3 jsonLog: %v", err)
 		}
+
+		// All done with this vote
+		// Vote completed
+		p.Lock()
+		p.ballotResults = append(p.ballotResults, *vr)
+		fmt.Printf("%v finished bunch %v vote %v -- "+
+			"total progress %v/%v\n", time.Now(), bunchID,
+			voteID, len(p.ballotResults), cap(p.ballotResults))
+		p.Unlock()
```

Comment:
Unlock needs to move above the Printf.

Reply:
Fixed.

Reply:
I reverted this change.  The print is required to be in the lock, otherwise a race occurs.  The tests caught this.